Packs/RecordedFuture/Integrations/RecordedFuture/RecordedFuture.py

```diff
@@ -1357,6 +1371,122 @@ def __build_links_context(self, links_data: Dict[str, List], markdown: str) -> C
                     outputs_key_field=""
                 )
 
+    def fetch_incidents(self, rule_names_arg: Optional[str], fetch_time: str) -> None:
+        if rule_names_arg:
+            rule_names = rule_names_arg.split(';')
+        else:
+            rule_names = []
+
+        if not fetch_time:
+            fetch_time = '24 hours'
+
+
+        last_run = demisto.getLastRun()
+        if not last_run:
+            last_run = {}
+        if 'time' not in last_run:
+            time, _ = parse_date_range(fetch_time, date_format='%Y-%m-%dT%H:%M:%S.%fZ')
+        else:
+            time = last_run['time']
+
+        current_time = datetime.strptime(time, '%Y-%m-%dT%H:%M:%S.%fZ')
+        triggered_time = '[{},)'.format(datetime.strftime(current_time, '%Y-%m-%d %H:%M:%S'))
+        max_time = current_time
+
+        rule_ids = []  # type: list
+
+        for rule in rule_names:
+            rules = self.client.get_alert_rules(rule)
+            if rules and 'data' in rules:
+                rule_ids += map(lambda r: r['id'], rules['data'].get('results', []))
+
+        all_alerts = []  # type: list
+        if rule_ids:
+            for rule_id in rule_ids:
+                params = {"alertRule": rule_id, "triggered": triggered_time, "status": "no-action"}
+                alerts = self.client.get_alerts(params)
+                if alerts and 'data' in alerts:
+                    all_alerts += alerts['data'].get('results', [])
+        else:
+            params = {"triggered": triggered_time, "status": "no-action"}
+            alerts = self.client.get_alerts(params)
+            if alerts and 'data' in alerts:
+                all_alerts += alerts['data'].get('results', [])
+
+        incidents = []
+        update_data = []
+        for alert in all_alerts:
+            alert_time = datetime.strptime(alert['triggered'], '%Y-%m-%dT%H:%M:%S.%fZ')
+            # The API returns also alerts that are triggered in the same time
+            if alert_time > current_time:
+                #Set alerts status to pending
+                update_data.append({"id": alert['id'], "status": "pending"})
```

Comment:
Why set the alert status to pending?

Reply:
That is how we want to handle alerts that have been pulled from our system.  Pending means that we are still waiting for users to interact with it.